torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -3583,10 +3583,29 @@ def _get_param_name_to_param(
     return dict(zip(param_to_param_name.values(), param_to_param_name.keys()))
 
 
-def clean_param_name(param_name: str) -> str:
-    """Cleans the parameter name by removing any FSDP-related prefixes."""
-    # FSDP full parameter names may not have both (i.e. `FSDP_PREFIX`), so we
+def _get_buffer_to_buffer_name(model: torch.nn.Module):
+    def module_fn(module, prefix, buffer_to_buffer_name):
+        if not isinstance(module, FullyShardedDataParallel):
+            for buffer_name, buffer in module.named_buffers(recurse=False):
+                is_shared_buffer = buffer in buffer_to_buffer_name
```

Comment:
can shared buffers have distinct names? If so, might be worth returning both of them, just to be extra sure we never clone buffers?